flink-tests/src/test/java/org/apache/flink/test/recovery/UpdateJobResourceRequirementsRecoveryITCase.java

```diff
@@ -0,0 +1,114 @@
+
+package org.apache.flink.test.recovery;
+
+import org.apache.flink.api.common.JobID;
+import org.apache.flink.client.ClientUtils;
+import org.apache.flink.client.program.rest.RestClusterClient;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.configuration.HighAvailabilityOptions;
+import org.apache.flink.configuration.JobManagerOptions;
+import org.apache.flink.configuration.WebOptions;
+import org.apache.flink.core.testutils.AllCallbackWrapper;
+import org.apache.flink.runtime.jobgraph.JobGraph;
+import org.apache.flink.runtime.jobgraph.JobGraphTestUtils;
+import org.apache.flink.runtime.jobgraph.JobResourceRequirements;
+import org.apache.flink.runtime.jobgraph.JobVertex;
+import org.apache.flink.runtime.minicluster.MiniCluster;
+import org.apache.flink.runtime.minicluster.MiniClusterConfiguration;
+import org.apache.flink.runtime.testtasks.BlockingNoOpInvokable;
+import org.apache.flink.runtime.zookeeper.ZooKeeperExtension;
+import org.apache.flink.test.scheduling.UpdateJobResourceRequirementsITCase;
+import org.apache.flink.util.TestLogger;
+import org.apache.flink.util.TestLoggerExtension;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.nio.file.Path;
+
+import static org.apache.flink.core.testutils.FlinkAssertions.assertThatFuture;
+
+/** Tests for recovering of rescaled jobs. */
+@ExtendWith(TestLoggerExtension.class)
+class UpdateJobResourceRequirementsRecoveryITCase extends TestLogger {
+
+    @RegisterExtension
+    private static final AllCallbackWrapper<ZooKeeperExtension> ZOOKEEPER_EXTENSION =
+            new AllCallbackWrapper<>(new ZooKeeperExtension());
+
+    /** Tests that a rescaled job graph will be recovered with the latest parallelism. */
+    @Test
+    void testRescaledJobGraphsWillBeRecoveredCorrectly(@TempDir Path tmpFolder) throws Exception {
+        final JobVertex jobVertex = new JobVertex("operator");
+        jobVertex.setParallelism(1);
+        jobVertex.setInvokableClass(BlockingNoOpInvokable.class);
+        final JobGraph jobGraph = JobGraphTestUtils.streamingJobGraph(jobVertex);
+        final JobID jobId = jobGraph.getJobID();
+
+        final Configuration configuration = new Configuration();
+        configuration.set(WebOptions.REFRESH_INTERVAL, 50L);
+        configuration.set(JobManagerOptions.SCHEDULER, JobManagerOptions.SchedulerType.Adaptive);
+        configuration.set(HighAvailabilityOptions.HA_MODE, "zookeeper");
+        configuration.set(
+                HighAvailabilityOptions.HA_ZOOKEEPER_QUORUM,
+                ZOOKEEPER_EXTENSION.getCustomExtension().getConnectString());
+        configuration.set(
+                HighAvailabilityOptions.HA_STORAGE_PATH, tmpFolder.toFile().getAbsolutePath());
+        Assertions.assertThat(tmpFolder.toFile()).isDirectory();
```

Comment:
nit: add a static import for `assertThat`?

Reply:
Why are we testing junit5 behavior?

Reply:
it's a leftover, I wasn't sure how junit behaves